### DISCUSSION TITLE
"previousComponent" is never assigned a value

### DISCUSSION THREAD

**Discussion by RRDAWLX** [preactjs/preact#3270] [Q&A] [answered]
```
/**
 * Keep track of the previous component so that we can set
 * `currentComponent` to `null` and throw when a hook is invoked
 * outside of render
 * @type {import('./internal').Component}
 */
let previousComponent;
```
The variable "previousComponent" in [hooks](https://github.com/preactjs/preact/blob/master/hooks/src/index.js) is designed to keep track of the previous component. But it is never assigned a value. How can it keep track of the previous component ? Is this a bug ?

**Comment by marvinhagemeister** [comment, marked as answer]
I wouldn't say it's a bug since it's an internal value that isn't part of any public API and setting or not setting it doesn't change the execution in any way for users. It will probably be eliminated by terser when compiling preact during the publish step.

Nonetheless, as you've found out it looks like dead weight and something we can remove. Would you be open to file a PR for that?

**Reply by RRDAWLX** [reply]
I've submitted a PR #3271
